htlcswitch/link.go

```diff
@@ -1230,7 +1247,9 @@ func (l *channelLink) htlcManager() {
 
 	// We've successfully reestablished the channel, mark it as such to
 	// allow the switch to forward HTLCs in the outbound direction.
+	l.Lock()
```

Comment:
seems unnecessary as `l.reestablished` is wrapped in atmoic?

Reply:
hmm I suppose you're right